xbmc/epg/EpgContainer.cpp

```diff
@@ -233,20 +234,27 @@ void CEpgContainer::LoadFromDB(void)
   m_bLoaded = bLoaded;
 }
 
-bool CEpgContainer::PersistTables(void)
+bool CEpgContainer::PersistTables(bool bPostOnly /* = true */)
 {
-  m_critSection.lock();
-  auto copy = m_epgs;
-  m_critSection.unlock();
+  std::map<unsigned int, CEpg*> copy;
+  {
+    CSingleLock lock(m_critSection);
+    if (bPostOnly) //Persist tables only if db is disabled as persistAll do already this job here.
+      return m_bCallPersistTables = !CSettings::Get().GetBool("epg.ignoredbforclient");
+    else
+      copy = m_epgs;
+  }
   return m_database.Persist(copy);
```

Comment:
after a second look it makes more sense to move out the stuff which marks it to persist in a separate method `MarkTablesToPersist()` or something similar and call this method in `CPVRChannelGroupInternal::CreateChannelEpgs`. There is only one place where this method is called and we don't need to adjust the `PersistTables`method in that case.